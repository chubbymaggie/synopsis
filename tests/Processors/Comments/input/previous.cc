// -*- Comments.SSFilter -*-
// -*- Comments.Previous -*-

struct Foo
{
  int first;  //< first comment
  int second; //< second comment
  int third;  //< third comment
};

enum Enum
{
  ONE,  //< ONE comment
  TWO,  //< TWO comment
  THREE //< THREE comment
};
