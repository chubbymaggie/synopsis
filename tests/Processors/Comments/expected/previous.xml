<?xml version='1.0' encoding='ISO-8859-1'?>
<ast>
 <declarations>
  <instance accessibility="0" class="Synopsis.AST.Class" file="Processors/Comments/input/previous.cc" line="4" name="Foo" type="struct">
   <declarations>
    <item>
     <instance accessibility="1" class="Synopsis.AST.Variable" constr="0" file="Processors/Comments/input/previous.cc" line="6" name="Foo.first" type="data member">
      <vtype>
       <instance class="Synopsis.Type.Base" language="C++" name="int"/>
      </vtype>
      <annotations>
       <key>
        comments
       </key>
       <value/>
      </annotations>
     </instance>
    </item>
    <item>
     <instance accessibility="1" class="Synopsis.AST.Variable" constr="0" file="Processors/Comments/input/previous.cc" line="7" name="Foo.second" type="data member">
      <vtype/>
      <annotations>
       <key>
        comments
       </key>
       <value>
        <item>
          second comment
        </item>
       </value>
      </annotations>
     </instance>
    </item>
    <item>
     <instance accessibility="1" class="Synopsis.AST.Variable" constr="0" file="Processors/Comments/input/previous.cc" line="8" name="Foo.third" type="data member">
      <vtype/>
      <annotations>
       <key>
        comments
       </key>
       <value>
        <item>
          third comment
        </item>
       </value>
      </annotations>
     </instance>
    </item>
    <item>
     <instance accessibility="1" class="Synopsis.AST.Builtin" file="Processors/Comments/input/previous.cc" line="8" name="EOS" type="EOS">
      <annotations>
       <key>
        comments
       </key>
       <value>
        <item>
         &lt; third comment
        </item>
       </value>
      </annotations>
     </instance>
    </item>
   </declarations>
   <annotations>
    <key>
     comments
    </key>
    <value>
     <item>
      -*- Comments.SSFilter -*-
-*- Comments.Previous -*-
     </item>
     <item/>
    </value>
   </annotations>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Enum" file="Processors/Comments/input/previous.cc" line="11" name="Enum" type="enum">
   <enumerators>
    <item>
     <instance accessibility="0" class="Synopsis.AST.Enumerator" file="Processors/Comments/input/previous.cc" line="13" name="ONE" type="enumerator" value="">
      <annotations>
       <key>
        comments
       </key>
       <value/>
      </annotations>
     </instance>
    </item>
    <item>
     <instance accessibility="0" class="Synopsis.AST.Enumerator" file="Processors/Comments/input/previous.cc" line="14" name="TWO" type="enumerator" value="">
      <annotations>
       <key>
        comments
       </key>
       <value>
        <item>
          TWO comment
        </item>
       </value>
      </annotations>
     </instance>
    </item>
    <item>
     <instance accessibility="0" class="Synopsis.AST.Enumerator" file="Processors/Comments/input/previous.cc" line="15" name="THREE" type="enumerator" value="">
      <annotations>
       <key>
        comments
       </key>
       <value>
        <item>
          THREE comment
        </item>
       </value>
      </annotations>
     </instance>
    </item>
   </enumerators>
   <annotations>
    <key>
     comments
    </key>
    <value/>
   </annotations>
   <eos>
    <instance accessibility="0" class="Synopsis.AST.Builtin" file="Processors/Comments/input/previous.cc" line="15" name="EOS" type="EOS">
     <annotations>
      <key>
       comments
      </key>
      <value>
       <item>
        &lt; THREE comment
       </item>
      </value>
     </annotations>
    </instance>
   </eos>
  </instance>
 </declarations>
 <types>
  <instance class="Synopsis.Type.Declared" language="C++" name="EOS">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="Enum">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="Foo">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="Foo.first">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="Foo.second">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="Foo.third">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="ONE">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="THREE">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="TWO">
   <declaration/>
  </instance>
 </types>
 <files>
  <instance class="Synopsis.SourceFile.SourceFile" language="C++" name="Processors/Comments/input/previous.cc" primary="1">
   <declarations>
    <item/>
    <item/>
    <item/>
    <item/>
    <item/>
    <item/>
   </declarations>
   <includes/>
   <macro_calls/>
  </instance>
 </files>
</ast>
